discussion/evaluation-of-project.tex

```diff
@@ -1,6 +1,80 @@
+
+A third option could have been to develop a suite of micro benchmarks.
+This option was not chosen as we wanted to create \emph{larger} projects to stress the programmability and code organization of the platform.
+
+\subsection{Performance}
+In \autoref{sec:res:perf} we looked at the performance of the graphical game application, Circle Game.
+It is evident, from our results, that the performance of the {\rust} version is comparable and even greater than that of equivalent {\C} code.
+Although a graphical game application and the \gls{fps} performance metric is not typical for the embedded processors targeted here, this qualitative result suggests that the abstractions provided by the Rust language are \emph{zero cost} when considering performance.
+
+\subsection{Code Size}
+
+From our code size analysis we have made three discoveries which are worth to discuss.
+This discussion should be read in light of the fact that the {\rust} compiler is a newly released compiler and is being compared to a {\C} compiler which have been used in production for years.
+
+\begin{description}
+
+\item [Large unoptimized {\rust} build] \hfill \\
+  As seen throughout all of our measurements in {\rust}, the unoptimized binaries are significantly larger than the optimized binaries.
+  The cause for this seems to be that the many abstractions of the {\rust} language are implemented by adding levels of indirection through function calls, like the for loop described in \autoref{sec:rust:loops}.
+  These \emph{extra} functions, which are eliminated when optimized, makes the compiler produce substantially larger binaries without the optimizations.
+
+\item [{\rust} code can be significantly larger than {\C} equivalents] \hfill \\
+  The optimized Sensor Tracker application written in {\rust} proved to be several times the size of the {\C} version.
+  This was further analyzed to reveal that the increase in size is due to the {\rust} libraries, especially considering generic code.
+  As an example, the Sensor Tracker will output samples over the \gls{usart} interface, these samples are converted to strings using the {\core} library \code{format!} macro.
+  Using this macro for the types \code{u8, i32, u32} increases the size of the optimized binary with 1736B.
+
+\item [There is zero size overhead of using {\rust}] \hfill \\
+  The Minimal Main application shows that using {\rust} by itself does not incur an increase in size.
+  It is the size of the software libraries that in our results increases the size of the {\rust} binaries
+
+\end{description}
+
+\subsection{Energy}
+
+We must emphasis that energy consumption measurements presented in \autoref{sec:res:energy} are preliminary and that the focus for the Sensor Tracker application was primarily to provide an application in this domain.
+The main problem with reading to much into these results, is that the {\rust} and {\C} implementations share a substantial amount of the library code, written in {\C}.
```

Comment:
Muligens legge på "which is written in C." på slutten av setningen? Vet ikke helt..